From eb39b5591fcace81a3d605b6d204468f9e690d14 Mon Sep 17 00:00:00 2001
From: Nico Weber <thakis@chromium.org>
Date: Thu, 13 May 2021 23:17:00 +0000
Subject: [PATCH] build: Convert tools/grit/preprocess_if_expr.gni to py3

Bug: 1205590
Change-Id: I6e1e268b07fbea50ecb0d7c03b68da825420946b
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/2893802
Auto-Submit: Nico Weber <thakis@chromium.org>
Commit-Queue: Dirk Pranke <dpranke@google.com>
Reviewed-by: Dirk Pranke <dpranke@google.com>
Cr-Commit-Position: refs/heads/master@{#882747}
---
 src/tools/grit/preprocess_if_expr.gni | 4 +---
 src/tools/grit/preprocess_if_expr.py  | 5 +++--
 2 files changed, 4 insertions(+), 5 deletions(-)

diff --git a/src/tools/grit/preprocess_if_expr.gni b/src/tools/grit/preprocess_if_expr.gni
index 71c1cc1e8d..64d21f9a3e 100644
--- a/src/tools/grit/preprocess_if_expr.gni
+++ b/src/tools/grit/preprocess_if_expr.gni
@@ -2,12 +2,10 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-import("//build/config/python.gni")
 import("//tools/grit/grit_defines.gni")
 
 template("preprocess_if_expr") {
-  # TODO(crbug.com/1112471): Get this to run cleanly under Python 3.
-  python2_action(target_name) {
+  action(target_name) {
     script = "//tools/grit/preprocess_if_expr.py"
 
     if (defined(invoker.deps)) {
diff --git a/src/tools/grit/preprocess_if_expr.py b/src/tools/grit/preprocess_if_expr.py
index b06735fde9..b8f09da4db 100644
--- a/src/tools/grit/preprocess_if_expr.py
+++ b/src/tools/grit/preprocess_if_expr.py
@@ -101,8 +101,9 @@ def main(argv):
     manifest_data = {}
     manifest_data['base_dir'] = '%s' % args.out_folder
     manifest_data['files'] = args.in_files
-    manifest_file = open(
-        os.path.normpath(os.path.join(_CWD, args.out_manifest)), 'wb')
+    manifest_file = io.open(
+        os.path.normpath(os.path.join(_CWD, args.out_manifest)), 'w',
+        encoding='utf-8', newline='\n')
     json.dump(manifest_data, manifest_file)
   return
 
